src/modules/product/product.service.js

```
@@ -199,6 +202,29 @@
   result.fields = formatFieldValues(result.fieldValues);
   delete result.fieldValues;
 
+  // Create default variation if none provided
+  if (!variations || !Array.isArray(variations) || variations.length === 0) {
+    // Create a default variation with the same price as the base product
+    await new ProductVariation({
+      product: addedProduct._id,
+      name: `${addedProduct.name} - Default`,
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)